woocommerce/woocommerce-android
# Privacy banner presentation logic

Closes: #8946


Corresponding FluxC PR: https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2737

### Description

This PR adds logic for PR presentation. For business requirements, please see attached issue.

Determine country logic is explained here: pe5pgL-2Qg-p2 (mixed approach)

### Testing instructions


#### TC1 WPCOM user from GDPR country
1. If needed, use VPN and change country to some European with GDPR. E.g. Germany, France, Spain
2. Install `trunk` (we want to test database migration)
3. Run the app
4. Install app from `issue/8946_privacy_banner_presentation`
5. Log in as WPCOM user
6. When `MyStore` is presented, make sure that the "Manage privacy" banner is shown
7. Dismiss the banner
8. Make sure that banner is no longer presented in the lifecycle of the app. It should be presented only at apps restart.

#### TC2 WPCOM user from non-GDPR country
1. If needed, use VPN to change to non GDPR country
2. Install app from `issue/8946_privacy_banner_presentation`
3. Log in as WPCOM user
4. When `MyStore` is presented, make sure that the "Manage privacy" banner is not shown

#### TC3 Non-WPCOM user from GDPR country
1. Make sure to remove SIM-card from the phone, if you have one (we want to test checking locale, not telephone carrier)
2. Change locale of phone to the one from GDPR country. See examples in TC1
3. Install app from `issue/8946_privacy_banner_presentation`
4. Log in via site credentials
5. Make sure that the "Manage privacy" banner is shown

#### TC4 Non-WPCOM user from nonGDPR country
1. Make sure to remove SIM-card from the phone, if you have one
2. Change locale of phone to the one from non-GDPR country
3. Install app from `issue/8946_privacy_banner_presentation`
4. Log in via site credentials
5. Make sure that the "Manage privacy" banner is not shown

### Images/gif

<img width="300" alt="image" src="https://github.com/woocommerce/woocommerce-android/assets/5845095/63586ae5-24fe-4c8b-adad-2277b21720ad">
